test/src/unit/template.test.js

```diff
@@ -412,6 +412,20 @@ describe('Templates.js', function() {
                 });
             });
         });
+        describe('(<validUrl>, <NoDesc> <NoName>)', function() {
+            it('succeeds, and gets the name and description from templates.json', async function() {
+                const url = 'https://raw.githubusercontent.com/kabanero-io/codewind-templates/master/devfiles/index.json';
+                const func = () => templateController.addRepository(url);
+                await (func().should.not.be.rejected);
+                templateController.repositoryList.should.deep.include({
```

Comment:
can this instead use an object from `templates.service.js`? (To reduce maintenance)